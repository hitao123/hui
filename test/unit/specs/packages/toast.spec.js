import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Toast from '../../../../packages/toast';
import ToastDemo from '../../../../docs/src/views/toast';

describe('Toast component', () => {
  it('should Loading as Circular', () => {
    const warpper = shallowMount(Toast, {
      propsData: {
        message: 'Toast',
        icon: 'wechat'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should Loading as Spinner', () => {
    expect(renderToString(ToastDemo)).toMatchSnapshot();
  });
});
