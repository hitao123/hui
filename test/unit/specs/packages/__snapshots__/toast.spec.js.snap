// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component should Loading as Circular 1`] = `
<div class="h-toast" style="display: none;" name="toast" icon="wechat">
  <!----> <span class="h-toast__text">Toast</span></div>
`;

exports[`Toast component should Loading as Spinner 1`] = `
<div><button class="h-button h-button--default h-button--large">
    <!----> <span class="h-button__text">click</span></button></div>
`;
